sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -204,11 +204,13 @@ export abstract class MessageReceiver extends LinkEntity<Receiver> {
     }
   }
 
-  protected createRheaLink(
+  protected async createRheaLink(
     options: ReceiverOptions,
     _abortSignal?: AbortSignalLike
   ): Promise<Receiver> {
-    return this._context.connection.createReceiver(options);
+    const receiver = await this._context.connection.createReceiver(options);
+    receiver.setMaxListeners(1000);
```

Comment:
Wouldn't the same be applicable to event-hubs?

And if possible, applying this logic at core-amqp might be more beneficial?

Reply:
Good point!

`Connection` lives in rhea-promise. I don't want to bump the limit there as other packages than EH and SB also depend on it.  We could extend `Connection` in core-amqp.